### PR TITLE
Update to fix JS error when 100% off coupon is applied

### LINKED ISSUE #1186: 100% coupon causes JavaScript error
If a coupon is added during checkout, WC Pay throws the following JavaScript error:

```
Uncaught IntegrationError: The selector you specified (#wcpay-card-element) applies to no DOM elements that are currently on the page.
Make sure the element exists on the page before calling mount().
    at new r (https://js.stripe.com/v3/?ver=3.0:1:6143)
    at fi.<anonymous> (https://js.stripe.com/v3/?ver=3.0:1:121807)
    at fi.mount (https://js.stripe.com/v3/?ver=3.0:1:22390)
    at HTMLBodyElement.<anonymous> (https://mywebsite.com/wp-content/plugins/woocommerce-payments/dist/checkout.js?ver=1.8.0:1:5943)
    at HTMLBodyElement.dispatch (https://mywebsite.com/wp-includes/js/jquery/jquery.min.js?ver=3.5.1:2:43090)
    at HTMLBodyElement.v.handle (https://mywebsite.com/wp-includes/js/jquery/jquery.min.js?ver=3.5.1:2:41074)
    at Object.trigger (https://mywebsite.com/wp-includes/js/jquery/jquery.min.js?ver=3.5.1:2:71513)
    at HTMLBodyElement.<anonymous> (https://mywebsite.com/wp-includes/js/jquery/jquery.min.js?ver=3.5.1:2:72108)
    at Function.each (https://mywebsite.com/wp-includes/js/jquery/jquery.min.js?ver=3.5.1:2:2976)
    at s.fn.init.each (https://mywebsite.com/wp-includes/js/jquery/jquery.min.js?ver=3.5.1:2:1454)
r @ ?ver=3.0:1
(anonymous) @ ?ver=3.0:1
(anonymous) @ ?ver=3.0:1
(anonymous) @ checkout.js?ver=1.8.0:1
dispatch @ jquery.min.js?ver=3.5.1:2
v.handle @ jquery.min.js?ver=3.5.1:2
trigger @ jquery.min.js?ver=3.5.1:2
(anonymous) @ jquery.min.js?ver=3.5.1:2
each @ jquery.min.js?ver=3.5.1:2
each @ jquery.min.js?ver=3.5.1:2
trigger @ jquery.min.js?ver=3.5.1:2
success @ checkout.min.js?ver=4.8.0:1
c @ jquery.min.js?ver=3.5.1:2
fireWith @ jquery.min.js?ver=3.5.1:2
l @ jquery.min.js?ver=3.5.1:2
(anonymous) @ jquery.min.js?ver=3.5.1:2
load (async)
send @ jquery.min.js?ver=3.5.1:2
ajax @ jquery.min.js?ver=3.5.1:2
s.ajax.s.ajax @ jquery-migrate.min.js?ver=3.3.2:2
update_checkout_action @ checkout.min.js?ver=4.8.0:1
setTimeout (async)
update_checkout @ checkout.min.js?ver=4.8.0:1
dispatch @ jquery.min.js?ver=3.5.1:2
v.handle @ jquery.min.js?ver=3.5.1:2
trigger @ jquery.min.js?ver=3.5.1:2
(anonymous) @ jquery.min.js?ver=3.5.1:2
each @ jquery.min.js?ver=3.5.1:2
each @ jquery.min.js?ver=3.5.1:2
trigger @ jquery.min.js?ver=3.5.1:2
success @ checkout.min.js?ver=4.8.0:1
c @ jquery.min.js?ver=3.5.1:2
fireWith @ jquery.min.js?ver=3.5.1:2
l @ jquery.min.js?ver=3.5.1:2
(anonymous) @ jquery.min.js?ver=3.5.1:2
load (async)
send @ jquery.min.js?ver=3.5.1:2
ajax @ jquery.min.js?ver=3.5.1:2
s.ajax.s.ajax @ jquery-migrate.min.js?ver=3.3.2:2
submit @ checkout.min.js?ver=4.8.0:1
dispatch @ jquery.min.js?ver=3.5.1:2
v.handle @ jquery.min.js?ver=3.5.1:2
```

*Reproducing the error*

* Create a 100% off coupon
* Add product(s) to the cart
* Proceed to Checkout
* Add the 100% coupon
* See the error in the console

The error only occurs when the coupon is added on the checkout page, so the cart total is changed to 0 after loading WC Pay.

This was reported on [the forums](https://wordpress.org/support/topic/bug-in-checkout-js/?view=all), where it was narrowed down to this line in `dist/checkout.js`:

`e(document.body).on("updated_checkout",function(){e("#wcpay-card-element").children().length || (u.unmount(), u.mount("#wcpay-card-element"))})`

They changed it to this, which cleared the error:

`e(document.body).on("updated_checkout",function(){if(e("#wcpay-card-element").length){e("#wcpay-card-element").children().length || (u.unmount(), u.mount("#wcpay-card-element"))}})`

I was able to reproduce the error on my test site, and updating that JS in the dist file did prevent the error.

### PR DESCRIPTION
**Pull request by dmallory42**
Fixes #1186 

#### Changes proposed in this Pull Request

*  Update for a JS error which occurs when a 100% off coupon is applied. This error is caused because of an event which triggers when checkout is updated and part of it tries to check the `.length` of the `children()` of `$('#wcpay-card-element')`. If there is no fee to be paid, this element disappears from the page which is why this error was occurring.

#### Testing instructions

* Create a 100% off coupon in Marketing -> Coupons.
* Apply the coupon during checkout of a product.
* Verify that on `trunk` a JavaScript error occurs in the console.
* Verify that on this branch, the error no longer appears.

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: client/checkout/classic/index.js [hunk 1]
```diff
@@ -44,6 +44,11 @@ jQuery( function ( $ ) {
 	// event for this. This part of the page can also reload based on changes to checkout details, so we call unmount
 	// first to ensure the card element is re-mounted correctly.
 	$( document.body ).on( 'updated_checkout', () => {
+		// If the card element selector doesn't exist, then do nothing (for example, when a 100% discount coupon is applied).
+		if ( ! $( '#wcpay-card-element' ).length ) {
```
Comment (kalessil) [inline client/checkout/classic/index.js:48]: I'd suggest to merge this into next condition: `if ( ! $( '#wcpay-card-element' ).length || $( '#wcpay-card-element' ).children().length)`, as they guarding the element mounting.

**Review by kalessil — CHANGES_REQUESTED** [review]

**Comment by kalessil** [comment]
@dmallory42 : looks like the branch needs to be rebased - GH complains about conflicts.

File: client/checkout/classic/index.js (hunk 1 above)
Comment (dmallory42) [reply client/checkout/classic/index.js]: That makes sense. I've updated the PR in 7cb98b161ad5cf715847c561066aa4c0c84f0fa2.

**Comment by dmallory42** [comment]
Thanks @kalessil, that's updated now. Let me know if you think it's good to go.

**Review by kalessil — APPROVED** [review]
LGTM, :shipit:

**Pull request merged**